Make November 2024 news public, add December template (#29154)
idaholab/moose: modules/doc/content/newsletter/2024/2024_11.md

types, and which can be operated on by each `ChainControl` object. This enables creating limiters, PID controllers, etc,
and chaining these operations, to control parameters in the input file.
This was inspired by the `ControlLogic` system in the thermal-hydraulics module.

## MOOSE Modules Changes

### Execution of MultiApps within the SIMPLE algorithm for solving the Navier Stokes equation

The [SIMPLE.md] executioner was reworked to use a `SolveObject`, `SIMPLESolve`, to perform the execution.
This allowed nesting the solution algorithm inside the fixed-point iteration loop with [MultiApps.md].
This enables converging multi-physics problems. The `SIMPLE` executioner is currently limited to
steady state simulations.

### Re-implementation of the finite volume face flux limiters

A new implementation of the finite volume face flux limiters was performed. This new implementation presents the
expected second order convergence across additional limiters, and adds the `Ventakakrishnan` limiter.
It was demonstrated on the classical scalar-advection dispersion case and showed reduced numerical dispersion.

> ## libMesh-level Changes
> 

Reviewer: ```suggestion
```

# MOOSE Newsletter (November 2024)

## MOOSE Improvements

> 

Reviewer 2: @tanoret @grmnptr @freiler @joshuahansel @loganharbour I added your contributions to the newseletter. Please feel free to edit

can now solve multiple solver systems, e.g. nonlinear or linear systems. A variable can be placed in a solver system
using the `solver_sys` parameter of the `Variable`.

The solve of multiple systems can be iterated using the [!param](/Executioner/Transient/multi_system_fixed_point) parameter.
A [Convergence object](syntax/Convergence/index.md) must then be specified to the `Executioner` to assess the convergence
of the multi-system fixed point iteration.

This support was also extended to the [Physics system](syntax/Physics/index.md), in which the `Physics` equations can be
set to be solved in their own solver system

### Interface for finding data files

Parameters with the string-derived type `DataFileName` will now search the application's data folder, as well as the
local simulation folder, for the files mentioned in the parameter. They will then return the absolute path to those files.
This is intended to simplify the inclusion of data in an application, data which can be referred to in the input files.

### New ChainControl system for handling complex/chained Control logic

> The new [ChainControl system](syntax/ChainControls/index.md) was added to MOOSE. It uses `ControlData` which can be of arbitrary
> types, and which can be operated on by each `ChainControl` object. This enables creating limiters, PID controllers, etc,
> and chaining these operations, to control parameters in the input file.
> This was inspired by the `ControlLogic` system in the thermal-hydraulics module.

Reviewer 3: ```suggestion
The new [ChainControl system](syntax/ChainControls/index.md) was added to MOOSE. It uses `ChainControlData` which can be of arbitrary
types, and which can be operated on by each `ChainControl` object. This enables creating limiters, PID controllers, etc.,
and chaining these operations, to control parameters in the input file.
This will replace the `ControlLogic` system in the thermal-hydraulics module after a transition period.
```

- Add `DMSwarmComputeMoments()`
- Add `DMSwarmPushCellDM()` and `DMSwarmPopCellDM()`
- Add `DMPlexTransformGetMatchStrata()` and `DMPlexTransformSetMatchStrata()`
- Deprecate `DMPlexSetGlobalToNaturalSF()` and `DMPlexGetGlobalToNaturalSF()` for existing
  `DMSetNaturalSF()` and `DMGetNaturalSF()`
- Add `-dm_plex_box_label_bd` to setup isoperiodicity when using `-dm_plex_box_label_bd`

## Bug Fixes and Minor Enhancements

- [Periodic boundary conditions](syntax/BCs/Periodic/index.md) are now more
  verbose. They will now output intermediate setup data in their [!param](/BCs/Periodic/AddPeriodicBCAction/auto_direction)
  mode, and will error with a helpful error message if used using conflicting parameters, or with a
  non-existing primary or secondary boundary.
- All `Sampler` vectorpostprocessors (VPPs) will now correctly detect and throw an error when used with
  vector and array variables, which they currently do not support. New `Sampler` VPPs could be created
  to support them.
- [ProjectionAux.md] can now handle 1D elements in a multi-dimensional mesh - for example, when
  using 1D thermal hydraulic channels in a multi-dimensional loop.

> - A [NumTimesSteps.md] postprocessor was added to count the number of time steps elapsed in a simulation.

Reviewer 2: ```suggestion
- A [NumTimeSteps.md] postprocessor was added to count the number of time steps elapsed in a simulation.
```

- Lower-D elements created using the [!param](/Mesh/FileMesh/build_all_side_lowerd_mesh) parameter are now
  separated in multiple blocks depending on the full-dimensional block they are located next to. However,

>   block-restriction of certain kernels to part of the lower-D blocks is not supported.

Reviewer 2: ```suggestion
  block-restriction of certain kernels to only part of the lower-D blocks is not supported.
```
